Mathlib/Algebra/Polynomial/SumIteratedDerivative.lean

```diff
@@ -0,0 +1,263 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.BigOperators
+import Mathlib.Algebra.Polynomial.Degree.Lemmas
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Algebra.Polynomial.RingDivision
+
+/-!
+# Sum of iterated derivatives
+-/
+
+open Finset
+open scoped Nat
+
+namespace Polynomial
+
+variable {R S : Type*}
+
+section Semiring
+
+variable [Semiring R] [Semiring S]
+
+/--
+Iterated derivatives as a finite support function.
+-/
+@[simps! apply_toFun]
+noncomputable def derivativeFinsupp : R[X] →ₗ[R] ℕ →₀ R[X] where
```

Comment:
I'm very curious if there's some way to make this morphism multiplicative by choosing the codomain as some variant of `AddMonoidAlgebra` instead of just a `Finsupp`; but that's definitely not in scope for this PR!